pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -73,959 +71,762 @@
+                    f"`Trainer(accelerator='ipu', precision={self._precision_flag!r})` is not supported."
+                )
+        if isinstance(self.accelerator, TPUAccelerator) and self._precision_flag == 64:
+            raise MisconfigurationException(
+                "`Trainer(accelerator='tpu', precision=64)` is not implemented."
+                " Please, open an issue in `https://github.com/PyTorchLightning/pytorch-lightning/issues`"
+                " requesting this feature."
             )
-        elif self.use_ddp:
-            use_slurm_ddp = self.use_ddp and self._is_slurm_managing_tasks()
-            use_torchelastic_ddp = self.use_ddp and TorchElasticEnvironment.detect()
-            use_kubeflow_ddp = self.use_ddp and KubeflowEnvironment.detect()
-            use_ddp_spawn = self._strategy_type == _StrategyType.DDP_SPAWN
-            use_ddp_cpu_spawn = use_ddp_spawn and self.use_cpu
-            use_tpu_spawn = self.use_tpu and self._strategy_type == _StrategyType.TPU_SPAWN
-            use_ddp_cpu_torch_elastic = use_ddp_cpu_spawn and TorchElasticEnvironment.detect()
-            use_ddp_cpu_kubeflow = use_ddp_cpu_spawn and KubeflowEnvironment.detect()
-            use_ddp_cpu_slurm = use_ddp_cpu_spawn and self._is_slurm_managing_tasks()
-            use_ddp_sharded = self._strategy_type == _StrategyType.DDP_SHARDED
-            use_ddp_sharded_spawn = self._strategy_type == _StrategyType.DDP_SHARDED_SPAWN
-            use_ddp_fully_sharded = self._strategy_type == _StrategyType.DDP_FULLY_SHARDED
-
-            if use_tpu_spawn:
-                ddp_strategy_cls = TPUSpawnStrategy
-            elif use_ddp_sharded:
-                ddp_strategy_cls = DDPShardedStrategy
-            elif use_ddp_sharded_spawn:
-                ddp_strategy_cls = DDPSpawnShardedStrategy
-            elif (
-                use_ddp_cpu_slurm
-                or use_slurm_ddp
-                or use_ddp_cpu_torch_elastic
-                or use_torchelastic_ddp
-                or use_kubeflow_ddp
-                or use_ddp_cpu_kubeflow
-            ):
-                ddp_strategy_cls = DDPStrategy
-            elif use_ddp_spawn or use_ddp_cpu_spawn:
-                ddp_strategy_cls = DDPSpawnStrategy
-            elif use_ddp_fully_sharded:
-                ddp_strategy_cls = DDPFullyShardedStrategy
-            else:
-                ddp_strategy_cls = DDPStrategy
-
-            plugin = ddp_strategy_cls(
-                parallel_devices=self.parallel_devices, cluster_environment=self.cluster_environment
+        if (
+            isinstance(self.accelerator, TPUAccelerator)
+            and isinstance(self._precision_flag, PrecisionPlugin)
+            and not isinstance(self._precision_flag, (TPUPrecisionPlugin, TPUBf16PrecisionPlugin))
+        ):
+            raise ValueError(
+                f"The `TPUAccelerator` can only be used with a `TPUPrecisionPlugin`," f" found: {self._precision_flag}."
```

Comment:
can this also be moved into the TPU strategy? it has both the accelerator & precision classes to assert on this.

Reply:
Added in #11449